### PR TITLE
`ChunkTag` and `ColorTag` Tag and Mech Examples

### PR DESCRIPTION
**Pull request by BreadcrumbIsTaken**
### Adds examples for `ChunkTag` tags and mechs and `ColorTag` tags.

Made sure to test all the examples and used past reviews to make sure it works well.

This is kinda a big PR so take your time 👍 

Minor explanations:
- For `ColorTag`, as you can't see the color in the meta website, I just narrated the `.rbg` value to avoid raw objects and to show the output as rgb
- `<ChunkTag.add>` and `<ChunkTag.sub>` use `<chunk[...]>` and not `<player.location.chunk>` to show how the `add` and `sub` tags work.
- `ChunkTag` mech `load` uses `<player.location.chunk.add[100,0]>` instead of `<player.location.chunk>` because if it uses the player's current chunk then it is probably already loaded.

###### 🎨🎨🎨

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java [hunk 1]
```diff
@@ -292,6 +292,9 @@ public static void register() {
         // @returns ChunkTag
         // @description
         // Returns the chunk with the specified coordinates added to it.
+        // @example
+        // # Notes the chunk with 10 added onto the X and Z axes of the chunk, making the coordinates "15,15,world".
+        // - note <chunk[5,5,world].add[10,10].cuboid> as:my_new_chunk_cuboid
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java]: The cuboid noting is a bit odd for demonstrating basic chunk usage. I'd encourage chunk-related features like `chunkload` command, ChunkTag mechs, ...

**Comment by mcmonkey4eva** [comment]
`<ChunkTag.add> and <ChunkTag.sub> use <chunk[...]> and not <player.location.chunk> to show how the add and sub tags work.
ChunkTag mech load uses <player.location.chunk.add[100,0]> instead of <player.location.chunk> because if it uses the player's current chunk then it is probably already loaded.`
These are backwards... add/sub are valuable for getting relative coordinates, and loading is most commonly going to be used with statically defined (or at least predictably defined) chunks

File: plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java [hunk 2]
```diff
@@ -338,6 +344,13 @@ public static void register() {
         // @returns ElementTag(Boolean)
         // @description
         // Returns true if the chunk has already been generated.
+        // @example
+        // # One way this can return "false" is if the player has been teleported to a location in the chunk while offline
+        // # to a chunk that has not been interacted or have not had a player nearby to generate it.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java]: If you have to explain the weird edge case where this example even applies, it's probably not a great example of the applicable tag. There are much more straightforward cases where this tag can be used.

File: plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java [hunk 3]
```diff
@@ -375,6 +399,10 @@ public static void register() {
         // @description
         // Returns a list of plugins that are keeping this chunk loaded.
         // This is related to the <@link command chunkload> command.
+        // @example
+        // # Narrates the list of plugin names keeping the player's chunk loaded formatted into readable format.
+        // # Example: "Plugins keeping your chunk loaded: Denizen and Citizens".
+        // - narrate "Plugins keeping your chunk loaded: <player.location.chunk.plugin_tickets.parse[name].formatted>"
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java]: I believe `parse[name]` is redundant here

File: plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java [hunk 4]
```diff
@@ -616,6 +694,9 @@ public static void register() {
         // @returns ListTag(LocationTag)
         // @description
         // Returns a list of the highest non-air surface blocks in the chunk.
+        // @example
+        // # Spawns a creeper at a random surface block and adds 2 on the Y axis to prevent the creeper from suffocating.
+        // - spawn creeper <player.location.chunk.surface_blocks.random.add[0,2,0]>
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java]: Shouldn't this be `.above`? (Simpler tag, and should only need 1 up not 2)

File: plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java [hunk 5]
```diff
@@ -719,6 +814,8 @@ public void adjust(Mechanism mechanism) {
         // Removes a chunk from memory.
         // @tags
         // <ChunkTag.is_loaded>
+        // @example
+        // - adjust <player.location.chunk> unload
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java]: I feel like unloading the player's current chunk would be an issue.

File: plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java [hunk 6]
```diff
@@ -290,6 +293,9 @@ public static void register() {
         // Returns the Red, Green, and Blue values of this ColorTag as an integer number, equivalent to an integer reparse of <@link tag ColorTag.hex>.
         // Highest order bits are red, then green, then lowest is blue.
         // This is a rare special case encoding usually avoided by most systems, but may be necessary for some obscure tools.
+        // @example
+        // # Narrates "16711935".
+        // - narrate <color[fuchsia].rgb_integer>
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java]: As said on the previous PR, if you don't know what a tag is for, don't jank an example of it.

File: plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java [hunk 7]
```diff
@@ -405,6 +441,9 @@ public static void register() {
         // @description
         // Returns the HSV value of this color.
         // EG, 100,100,255
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java]: Should remove the pre-example-meta examples ^

File: plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java [hunk 8]
```diff
@@ -446,6 +494,9 @@ public static void register() {
         // @returns ColorTag
         // @description
         // Returns a copy of this color object with a different alpha value (0 to 255).
+        // @example
+        // # Colors the text with a rgb value of "255,0,255,150", which is fuchsia, but transparency does not show up in chat.
+        // - narrate "<&color[<color[fuchsia].with_alpha[150]>]>This is fuchsia with a different alpha value! It does not show in chat though!"
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java]: If the example doesn't work, why did you commit it?

File: plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java [hunk 9]
```diff
@@ -502,6 +568,9 @@ public static void register() {
         // @returns ColorTag
         // @description
         // Returns the color that results if you mix this color with another.
+        // @example
+        // # Narrates "127,127,255", a mix between fuchsia and aqua which appears as a dark purple.
+        // - narrate <color[fuchsia].mix[aqua].rgb>
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java]: This example would be better as an `<&color[...` style one

**Comment by BreadcrumbIsTaken** [comment]
I just decided to remove the `<ColorTag.rgb_integer>` and `<ColorTag.rgba_integer>` examples, because like you said in your review, I wasn't sure what they were supposed to do 😅. Sorry for the 'jank' example I gave for those :)

File: plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java [hunk 10]
```diff
@@ -292,6 +292,9 @@ public static void register() {
         // @returns ChunkTag
         // @description
         // Returns the chunk with the specified coordinates added to it.
+        // @example
+        // # Loads the chunk with 10 added onto the X and Z axes of the player's chunk, making the coordinates "15,15,world".
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ChunkTag.java]: Description doesn't make sense anymore. `sub` example below is also borked.

File: plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java [hunk 11]
```diff
@@ -480,6 +528,9 @@ public static void register() {
         // @returns ColorTag
         // @description
         // Returns a copy of this color object with a different brightness value (0 to 255).
+        // @example
+        // # Colors the text with a rgb value of "150,0,148", which is a purple.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/ColorTag.java]: Extremely minor nitpick but that is not purple, that's magenta.

**Comment by BreadcrumbIsTaken** [comment]
Pushed fixes 👍

**Pull request merged**